async-nats/src/lib.rs

```diff
@@ -773,6 +797,26 @@ impl ConnectionHandler {
             Command::Flush { observer } => {
                 self.flush_observers.push(observer);
             }
+            Command::Drain { sid } => {
+                let mut drain_sub = |sid: u64, sub: &mut Subscription| {
+                    sub.is_draining = true;
+                    self.connection
+                        .enqueue_write_op(&ClientOp::Unsubscribe { sid, max: None });
+                };
+
+                if let Some(sid) = sid {
+                    if let Some(sub) = self.subscriptions.get_mut(&sid) {
+                        drain_sub(sid, sub);
+                    }
+                } else {
+                    // sid isn't set, so drain the whole client
+                    self.connector.events_tx.try_send(Event::Draining).ok();
```

Comment:
Assumed it was fine to ignore any errors here as they'd be pretty unlikely (the only scenario I could think of would be if the drain command was sent mid-shutdown, in which case it's fine to just ignore the error).